src/db/commit.rs

```diff
@@ -180,6 +180,22 @@ impl Commit {
         let chunk = dag::Chunk::new(builder.collapse(), &[value_hash]);
         Commit { chunk }
     }
+
+    pub async fn base_snapshot(
+        hash: &str,
+        dag_read: &dag::Read<'_>,
+    ) -> Result<Commit, FromHashError> {
```

Comment:
Define a separate enum containing errors for this function: BaseSnapshotError. Instead of ChunkMissing use something like NoBasisHash(String) where the String parameter is the hash of the chunk lacking a basis.